Compilation errors building with mariadb >= 10.3.28

The following compile errors occur when building this storage engine with mariadb >= 10.3.28. These errors look to originate from [21809f9a450df1bc44cef36377f96b516ac4a9ae](https://github.com/MariaDB/server/commit/21809f9a450df1bc44cef36377f96b516ac4a9ae) that changes the following function prototypes:
    
* tmp_use_all_columns, dbug_tmp_use_all_columns
  to accept MY_BITMAP** and to return MY_BITMAP * instead of my_bitmap_map*

* tmp_restore_column_map, dbug_tmp_restore_column_maps to accept
  MY_BITMAP* instead of my_bitmap_map*


```
~/mariadb-10.3.29/storage/ibmdb2i/db2i_conversion.cc:244:88: error: cannot convert 'MY_BITMAP* {aka st_bitmap*}' to 'MY_BITMAP** {aka st_bitmap**}' for argument '2' to 'MY_BITMAP* tmp_use_all_columns(TABLE*, MY_BITMAP**)'
       my_bitmap_map *old_map = tmp_use_all_columns(field->table, field->table->read_set);
                                                                                        ^
~/mariadb-10.3.29/storage/ibmdb2i/db2i_conversion.cc:376:61: error: cannot convert 'MY_BITMAP* {aka st_bitmap*}' to 'MY_BITMAP** {aka st_bitmap**}' for argument '1' to 'void tmp_restore_column_map(MY_BITMAP**, MY_BITMAP*)'
       tmp_restore_column_map(field->table->read_set, old_map);
                                                            ^

```


```
~/mariadb-10.3.29/storage/ibmdb2i/ha_ibmdb2i.cc:853:74: error: no matching function for call to 'dbug_tmp_use_all_columns(TABLE*&, MY_BITMAP*&)'
   my_bitmap_map *old_map= dbug_tmp_use_all_columns(table, table->read_set);
                                                                          ^
In file included from ~/mariadb-10.3.29/storage/ibmdb2i/../../sql/sql_show.h:22:0,
                 from ~/mariadb-10.3.29/storage/ibmdb2i/db2i_global.h:62,
                 from ~/mariadb-10.3.29/storage/ibmdb2i/ha_ibmdb2i.h:52,
                 from ~/mariadb-10.3.29/storage/ibmdb2i/ha_ibmdb2i.cc:53:
~/mariadb-10.3.29/storage/ibmdb2i/../../sql/table.h:3007:26: note: candidate: MY_BITMAP* dbug_tmp_use_all_columns(TABLE*, MY_BITMAP**)
 static inline MY_BITMAP *dbug_tmp_use_all_columns(TABLE *table,
                          ^~~~~~~~~~~~~~~~~~~~~~~~
~/mariadb-10.3.29/storage/ibmdb2i/../../sql/table.h:3007:26: note:   no known conversion for argument 2 from 'MY_BITMAP* {aka st_bitmap*}' to 'MY_BITMAP** {aka st_bitmap**}'
~/mariadb-10.3.29/storage/ibmdb2i/../../sql/table.h:3030:20: note: candidate: void dbug_tmp_use_all_columns(TABLE*, MY_BITMAP**, MY_BITMAP**, MY_BITMAP**)
 static inline void dbug_tmp_use_all_columns(TABLE *table,
                    ^~~~~~~~~~~~~~~~~~~~~~~~
~/mariadb-10.3.29/storage/ibmdb2i/../../sql/table.h:3030:20: note:   candidate expects 4 arguments, 2 provided
~/mariadb-10.3.29/storage/ibmdb2i/ha_ibmdb2i.cc:892:55: error: cannot convert 'MY_BITMAP* {aka st_bitmap*}' to 'MY_BITMAP** {aka st_bitmap**}' for argument '1' to 'void dbug_tmp_restore_column_map(MY_BITMAP**, MY_BITMAP*)'
   dbug_tmp_restore_column_map(table->read_set, old_map);
                                                       ^
~/mariadb-10.3.29/storage/ibmdb2i/ha_ibmdb2i.cc: In member function 'int32 ha_ibmdb2i::mungeDB2row(uchar*, const char*, const char*, bool)':
~/mariadb-10.3.29/storage/ibmdb2i/ha_ibmdb2i.cc:1293:81: error: no matching function for call to 'dbug_tmp_use_all_columns(TABLE*&, MY_BITMAP*&)'
   my_bitmap_map *old_write_map= dbug_tmp_use_all_columns(table, table->write_set);
                                                                                 ^
In file included from ~/mariadb-10.3.29/storage/ibmdb2i/../../sql/sql_show.h:22:0,
                 from ~/mariadb-10.3.29/storage/ibmdb2i/db2i_global.h:62,
                 from ~/mariadb-10.3.29/storage/ibmdb2i/ha_ibmdb2i.h:52,
                 from ~/mariadb-10.3.29/storage/ibmdb2i/ha_ibmdb2i.cc:53:
~/mariadb-10.3.29/storage/ibmdb2i/../../sql/table.h:3007:26: note: candidate: MY_BITMAP* dbug_tmp_use_all_columns(TABLE*, MY_BITMAP**)
 static inline MY_BITMAP *dbug_tmp_use_all_columns(TABLE *table,
                          ^~~~~~~~~~~~~~~~~~~~~~~~
~/mariadb-10.3.29/storage/ibmdb2i/../../sql/table.h:3007:26: note:   no known conversion for argument 2 from 'MY_BITMAP* {aka st_bitmap*}' to 'MY_BITMAP** {aka st_bitmap**}'
~/mariadb-10.3.29/storage/ibmdb2i/../../sql/table.h:3030:20: note: candidate: void dbug_tmp_use_all_columns(TABLE*, MY_BITMAP**, MY_BITMAP**, MY_BITMAP**)
 static inline void dbug_tmp_use_all_columns(TABLE *table,
                    ^~~~~~~~~~~~~~~~~~~~~~~~
~/mariadb-10.3.29/storage/ibmdb2i/../../sql/table.h:3030:20: note:   candidate expects 4 arguments, 2 provided
~/mariadb-10.3.29/storage/ibmdb2i/ha_ibmdb2i.cc:1297:62: error: cannot convert 'MY_BITMAP* {aka st_bitmap*}' to 'MY_BITMAP** {aka st_bitmap**}' for argument '2' to 'MY_BITMAP* tmp_use_all_columns(TABLE*, MY_BITMAP**)'
     old_read_map = tmp_use_all_columns(table, table->read_set);
                                                              ^
~/mariadb-10.3.29/storage/ibmdb2i/ha_ibmdb2i.cc:1324:57: error: cannot convert 'MY_BITMAP* {aka st_bitmap*}' to 'MY_BITMAP** {aka st_bitmap**}' for argument '1' to 'void tmp_restore_column_map(MY_BITMAP**, MY_BITMAP*)'
     tmp_restore_column_map(table->read_set, old_read_map);
                                                         ^
~/mariadb-10.3.29/storage/ibmdb2i/ha_ibmdb2i.cc:1325:62: error: cannot convert 'MY_BITMAP* {aka st_bitmap*}' to 'MY_BITMAP** {aka st_bitmap**}' for argument '1' to 'void dbug_tmp_restore_column_map(MY_BITMAP**, MY_BITMAP*)'
   dbug_tmp_restore_column_map(table->write_set, old_write_map);
                                                              ^
make[2]: *** [storage/ibmdb2i/CMakeFiles/ibmdb2i.dir/build.make:423: storage/ibmdb2i/CMakeFiles/ibmdb2i.dir/ha_ibmdb2i.cc.o] Error 1
```

--- a/db2i_conversion.cc
+++ b/db2i_conversion.cc
@@ -241,7 +241,11 @@ static void get_field_default_value(Field *field,
     defaultClause.append(" DEFAULT ");
     if (!field->is_null())
     {
-      my_bitmap_map *old_map = tmp_use_all_columns(field->table, field->table->read_set);
+      #if MYSQL_VERSION_ID >= 100328
+        MY_BITMAP *old_map = tmp_use_all_columns(field->table, &field->table->read_set);
+      #else
+        my_bitmap_map *old_map = tmp_use_all_columns(field->table, field->table->read_set);
+      #endif
       char tmp[MAX_FIELD_WIDTH];
       
       if (field->real_type() == MYSQL_TYPE_ENUM ||
@@ -373,7 +377,11 @@ static void get_field_default_value(Field *field,
         else
           defaultClause.length(0);
       }
-      tmp_restore_column_map(field->table->read_set, old_map);
+      #if MYSQL_VERSION_ID >= 100328
+        tmp_restore_column_map(&field->table->read_set, old_map);
+      #else
+        tmp_restore_column_map(field->table->read_set, old_map);
+      #endif
     }
     else if (field->maybe_null())
       defaultClause.append(STRING_WITH_LEN("NULL"));

--- a/ha_ibmdb2i.cc
+++ b/ha_ibmdb2i.cc
@@ -849,8 +849,11 @@ int ha_ibmdb2i::prepareRowForWrite(char* data, char* nulls, bool honorIdentCols)
   default_identity_value = FALSE;  
     
   ulong sql_mode = ha_thd()->variables.sql_mode;
-  
-  my_bitmap_map *old_map= dbug_tmp_use_all_columns(table, table->read_set);
+  #if MYSQL_VERSION_ID >= 100328
+    MY_BITMAP *old_map= dbug_tmp_use_all_columns(table, &table->read_set);
+  #else
+    my_bitmap_map *old_map= dbug_tmp_use_all_columns(table, table->read_set);
+  #endif
   for (Field **field = table->field; *field && !rc; ++field)
   {  
     int fieldIndex = (*field)->field_index;
@@ -888,8 +891,11 @@ int ha_ibmdb2i::prepareRowForWrite(char* data, char* nulls, bool honorIdentCols)
   if (!rc && db2Table->hasBlobs())
     rc = db2i_ileBridge::getBridgeForThread()->objectOverride(activeHandle,
                                                             activeWriteBuf->ptr());  
-
-  dbug_tmp_restore_column_map(table->read_set, old_map);
+  #if MYSQL_VERSION_ID >= 100328
+    dbug_tmp_restore_column_map(&table->read_set, old_map);
+  #else
+    dbug_tmp_restore_column_map(table->read_set, old_map);
+  #endif
 
   return rc; 
 }
@@ -1289,12 +1295,21 @@ int ha_ibmdb2i::rnd_end()
 int32 ha_ibmdb2i::mungeDB2row(uchar* record, const char* dataPtr, const char* nullMapPtr, bool skipLOBs)
 {
   DBUG_ASSERT(dataPtr);
-  
+  #if MYSQL_VERSION_ID >= 100328
+  MY_BITMAP *old_write_map= dbug_tmp_use_all_columns(table, &table->write_set);
+  MY_BITMAP *old_read_map;
+  #else
   my_bitmap_map *old_write_map= dbug_tmp_use_all_columns(table, table->write_set);
   my_bitmap_map *old_read_map;
+  #endif
   
-  if (unlikely(readAllColumns))
-    old_read_map = tmp_use_all_columns(table, table->read_set);
+  if (unlikely(readAllColumns)){
+    #if MYSQL_VERSION_ID >= 100328
+      old_read_map = tmp_use_all_columns(table, &table->read_set);
+    #else
+      old_read_map = tmp_use_all_columns(table, table->read_set);
+    #endif
+  }
 
   resetCharacterConversionBuffers();
   
@@ -1320,9 +1335,18 @@ int32 ha_ibmdb2i::mungeDB2row(uchar* record, const char* dataPtr, const char* nu
     
   }
     
-  if (unlikely(readAllColumns))
-    tmp_restore_column_map(table->read_set, old_read_map);
-  dbug_tmp_restore_column_map(table->write_set, old_write_map);
+  if (unlikely(readAllColumns)) {
+    #if MYSQL_VERSION_ID >= 100328
+      tmp_restore_column_map(&table->read_set, old_read_map);
+    #else
+      tmp_restore_column_map(table->read_set, old_read_map);
+    #endif
+  }
+  #if MYSQL_VERSION_ID >= 100328
+    dbug_tmp_restore_column_map(&table->write_set, old_write_map);
+  #else
+    dbug_tmp_restore_column_map(table->write_set, old_write_map);
+  #endif
   
   return 0;
 }

